pkgs/development/python-modules/manticore/default.nix

```diff
@@ -0,0 +1,109 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, capstone
+, crytic-compile
+, ply
+, prettytable
+, pyelftools
+, pyevmasm
+, pysha3
+, pyyaml
+, rlp
+, stdenv
+, unicorn
+, wasm
+, yices
+, pytestCheckHook
+, z3
+}:
+
+buildPythonPackage rec {
+  pname = "manticore";
+  version = "0.3.4";
+
+  src = fetchFromGitHub {
+    owner = "trailofbits";
+    repo = "manticore";
+    rev = version;
+    sha256 = "0i9653013wvnj6y7f7i7vyj97250wd1qspr9ylpb1pnhi3c23gi0";
+  };
+
+  propagatedBuildInputs = [
+    crytic-compile
+    ply
+    prettytable
+    pyevmasm
+    pysha3
+    pyyaml
+    rlp
+    wasm
+  ] ++ lib.optionals (stdenv.isLinux) [
+    capstone
+    pyelftools
+    unicorn
+  ];
+
+  # Python API is not used in the code, only z3 from PATH
```

Comment:
You will want to patch the z3 call sites to explicitly reference the z3 utility. Otherwise there's not guarantee that it will work.

Reply:
Not sure if I understand. The program is constructed in a way that it looks for z3 on PATH (or other solver, there is support for multiple) so if you want to use manticore with z3 you would do `nix-shell -p python3Packages.manticore z3`. The `z3-solver` dependency is not referenced anywhere in the code, it just uses the fact the `z3-solver` will place `z3` on PATH.